### PR TITLE
Remove in-memory trie backend.

### PR DESCRIPTION
**Pull request by arkpar**
Use regular trie backend with in-memory storage instead.

### CODE REVIEW

**Comment by arkpar** [comment]
`gitlab-test-browser-node` failure is unrelated

File: frame/executive/src/lib.rs [hunk 1]
```diff
@@ -615,7 +615,7 @@ mod tests {
 				header: Header {
 					parent_hash: [69u8; 32].into(),
 					number: 1,
-					state_root: hex!("489ae9b57a19bb4733a264dc64bbcae9b140a904657a681ed3bb5fbbe8cf412b").into(),
+					state_root: hex!("409fb5a14aeb8b8c59258b503396a56dee45a0ee28a78de3e622db957425e275").into(),
```
Comment (NikVolf) [inline frame/executive/src/lib.rs:618]: Why have this changed?

Comment (arkpar) [reply frame/executive/src/lib.rs:618]: Previous implementation of in memory backend used `trie_root` function which allows empty values in the trie. `TrieDBMut` that's used by the regular backend skips all empty values, hence storage root is different.

File: primitives/state-machine/src/in_memory_backend.rs [hunk 2]
```diff
@@ -19,134 +19,178 @@
...
-			inner: Default::default(),
-			trie: None,
-			_hasher: PhantomData,
+	/// Insert values into backend trie.
+	pub fn insert<
+		T: IntoIterator<Item = (Option<ChildInfo>, StorageCollection)>
+	>(
+		&mut self,
+		changes: T,
+	) {
```
Comment (NikVolf) [inline primitives/state-machine/src/in_memory_backend.rs:94]:
```suggestion
	pub fn insert(
		&mut self,
		changes: impl IntoIterator<Item = (Option<ChildInfo>, StorageCollection)>,
	) {
```

File: primitives/state-machine/src/in_memory_backend.rs [hunk 3]
```diff
@@ -19,134 +19,178 @@
...
+where
+	H::Out: Codec + Ord,
+{
+	/// Copy the state, with applied updates
+	pub fn update<
+		T: IntoIterator<Item = (Option<ChildInfo>, StorageCollection)>
+	>(
+		&self,
+		changes: T,
+	) -> Self {
```
Comment (NikVolf) [inline primitives/state-machine/src/in_memory_backend.rs:82]: pub fn update(
		&self,
		changes: impl IntoIterator<Item = (Option<ChildInfo>, StorageCollection)>,
	) -> Self {

Comment (arkpar) [reply primitives/state-machine/src/in_memory_backend.rs:82]: I'd prefer to stick to the style that's already adopted by the crate. Such changes should be applied with `cargo fix` or some other tool on a per-crate basis.

File: primitives/state-machine/src/in_memory_backend.rs (hunk 2 above)
Comment (arkpar) [reply primitives/state-machine/src/in_memory_backend.rs:94]: I'd prefer to stick to the style that's already adopted by the crate. Such changes should be applied with `cargo fix` or some other tool on a per-crate basis.

Comment (NikVolf) [reply primitives/state-machine/src/in_memory_backend.rs:94]: They will never be applied by cargo fix I think, it probably requires compiler context

**Pull request merged**